since-3.7.py

```diff
@@ -1,4 +1,3 @@
-# XXX Beware that if you use type hints,
-# this will become no-op only in Python 3.11.
-
-from __future__ import annotations  # Python >= 3.7 is required
+if 0:
+    async def f(x):
```

Comment:
(Perhaps this line should also be commented, for Python versions too old to support `async` at all? And similar in `since-3.11.py`?)

Reply:
Good catch.
Instead of adding another comment, maybe we could put everything on one line? Not sure.
Either way, I should fix the test suite to make sure that the "Python >= X.Y is required" comment actually shows up in the traceback.

Reply:
This turned out to be a bit more involved than I expected…
`since-3.7` fixed in 329982171fa396f60a292d4633e022987162c62f.
`since-3.11` fixed in 58b52e7256c13c71e382b4a0cbbcf95345df3088.
Tests fixed in 40a32e4e336d48e7e84462a6b075fa4d43d7d028 (except for the tricky case of Python 3.10, which I hope to fix later).